docs/proposals/auth-gateway.md

```diff
@@ -0,0 +1,57 @@
+---
+title: "Authentication Gateway"
+linkTitle: "Authentication Gateway"
+weight: 1
+slug: "auth-gateway"
+---
+
+- Author: [Doğukan Teber](https://github.com/dogukanteber)
+- Date: March 2023
+- Status: Proposed
+
+## Overview
+
+If you run Cortex for multiple tenants you need to identify your tenants every time they send metrics or query them. This is needed to ensure that metrics can be ingested and queried separately from each other. For this purpose, the Cortex microservices require you to pass a Header called `X-Scope-OrgID`. Unfortunately, the Prometheus Remote write API has no config option to send headers and for Grafana you must provide a data source to do so. Therefore the Cortex k8s manifests suggest deploying an NGINX server inside of each tenant which acts as a reverse proxy. Its sole purpose is proxying the traffic and setting the `X-Scope-OrgID` header for your tenant.
```

Comment:
and route the request to the right microservice? IE: distributors, query-frontend, ruler...

Reply:
Yes, exactly